### ISSUE TITLE
Allow configuration to remove non-expanded parameter sets

### ISSUE THREAD

**Issue by markcowl** [Azure/autorest.powershell#354]
For `POST` operations in ARM APIs, when there is no body parameter, the generated parameter sets work very well.

When there is a body parameter, there is no value in providing the body parameter except as expanded parameters - for instance, there i no piping into this body parameter that would make sense in this case.

Configuration allows removing the non-expanded parameter sets when they exist, however there is no easy way to supply a directive that removes non-expanded parameter sets **where expanded parameter sets exist**.  And this will be *very* common in `POST` scenarios.

So, we should at least have a directive that allows removing non-expanded parameter sets (and does nothign if no expanded parameter sets exist).

We should consider removing these parameter sets by default for all POST operations in Azure.

**Comment by fearthecowboy** [comment]
This can be done in modifiers plugin -- just add a way of removing the selected cmdlet variants.

**Comment by fearthecowboy** [comment]
```
directive:
  where:  ..
  removeNonExpanded: true
```

**Comment by dolauli** [comment]
p4

**Comment by dolauli** [comment]
Agreed "We should consider removing these parameter sets by default for all POST operations in Azure."
